src/components/Label/Label.tsx

```diff
@@ -43,11 +68,57 @@ class Label extends UIComponent<any, any> {
 
   static variables = labelVariables
 
+  handleIconOverrides = predefinedProps => ({
+    onClick: e => {
+      _.invoke(predefinedProps, 'onClick', e)
+      _.invoke(this.props, 'onIconClick', e, this.props)
+    },
+  })
+
   renderComponent({ ElementType, classes, rest }) {
-    const { children, content } = this.props
+    const { children, content, icon, iconPosition, onIconClick } = this.props
+    const getContent = (): ReactNode => {
+      if (childrenExist(children)) {
+        return children
+      }
+
+      const iconIsAfterContent = iconPosition === 'after'
+
+      const iconProps = {
+        className: classes.icon,
+        ...(icon &&
+          typeof icon === 'string' && {
+            name: icon,
+            ...(onIconClick && { tabIndex: '0' }),
+            variables: { color: classes.root.color },
```

Comment:
I agree on this, it would be much better. Just to be clear that we are on the same page: I will introduce the color for the label as a color variable inside the label variables, and provide it to the icon in the following manner:
```
variables: { color: Label.variables().color },
```
If this is the change that we both aim to, I would rather introduce it in this PR.